R/layer.r

```diff
@@ -345,14 +345,15 @@ Layer <- ggproto("Layer", NULL,
     self$position$compute_layer(data, params, layout)
   },
 
-  compute_geom_2 = function(self, data) {
-    # Combine aesthetics, defaults, & params
+  compute_geom_2 = function(self, data, plot) {
     if (empty(data)) return(data)
 
+    # evaluate defaults for theme
+    defaults <- self$geom$eval_defaults(theme = plot$theme)
```

Comment:
I'm not entirely sure that `plot$theme` is the complete theme when this is called. This needs to be verified.

Reply:
Can confirm it's not! Which explains a lot of the test failures. Will fix.